### PR TITLE
#130 chatforyou harness system 확장

### PR DESCRIPTION
**Pull request by SeJonJ**
ChatForYou 의 harness system 확장을 위한 수정 내용
1. claude code - codex 의 agent, skill, hook 등에서 서로 다른 리소스 문제가 발생.
- 이를 해결하기 위해 claude code 와 codex 의 각 리소스 정합성 교정
-> 추후 SAGE 프로젝트 적용으로 모든 리소스에 대해 SSOT 진행 예정

2. 보다 상세한 커밋은 아래 커밋 참고 : 메인 리소스와 merge 하느라 충돌 발생하여 수정함
- ed4567b5

## Summary by Sourcery

빌드/테스트/문법 검증을 위한 결정적(deterministic) 검증 게이트를 도입하고, 이를 중심으로 Claudex/Codex 훅 동작과 문서를 맞추며, 에이전트 간 하네스(harness) 동작이 일관되게 유지되도록 회귀 테스트와 CI를 추가합니다.

Enhancements:
- WebRTC 변경 사항에 대해 더 강력한 매칭 시그널과 더 명확한 게이트 메시지를 제공하는, 기능(feature)/플랜(plan) 인식 L3 리뷰 감지 기능을 추가합니다.
- 하드코딩된 로컬 경로 대신 git을 통해 프로젝트 루트를 찾도록 하여 Claude 및 Codex 훅을 이식성 있게 만듭니다.
- 에이전트 가이드와 WebRTC 리뷰 문서에 결정적 검증 프로토콜을 문서화하고 강제하며, 원칙(principle) 수준 결함을 다운그레이드하는 행위에 대해 더 엄격한 규칙을 추가합니다.
- 외부 전문가의 심각도(severity) 처리 규칙을 명확히 하여, 중대한 문제에 대한 정당하지 않은 다운그레이드를 방지합니다.

Build:
- 변경된 컴포넌트를 자동으로 감지하고, 위험도 수준에 따라 적절한 백엔드/프론트엔드 검증 명령을 실행하며, 재사용 가능한 검증 증적(evidence) 블록을 출력하는 `scripts/verify-changes.sh`를 추가합니다.
- `verify-changes.sh`를 위한 테스트 하네스 스크립트와, 변경 없음(no-change)·문서만 변경(docs-only)·L1 시나리오를 검증하기 위한 훅 테스트 샌드박스/픽스처를 추가합니다.

CI:
- 푸시 및 풀 리퀘스트 시 Claude/Codex 훅 회귀 테스트 스위트와 `verify-changes.sh` 스모크 테스트를 실행하는 GitHub Actions 워크플로를 도입합니다.

Documentation:
- 전용 검증 프로토콜 문서를 추가하고, 에이전트/스킬 가이드와 WebRTC 리뷰 프로토콜을 업데이트하여 Phase 03 종료 시점과 Phase 05 진입 시점에 결정적 검증을 요구하고, 단계 진행 또는 TODO 체크 전 증적을 기록하도록 규정합니다.

Tests:
- 플랜/리뷰 감지, Phase 4 체크리스트 완료, 선언된 위험도/세션 로그, 검증 동작의 엣지 케이스를 다루기 위해 Claude 및 Codex 훅 테스트 하네스를 새로운 샌드박스와 픽스처로 확장합니다.

Chores:
- 심각도 다운그레이드 제한 및 원칙 수준 결함 처리에 관해 Claude 및 Codex 에이전트 간 ChatForYou 외부 전문가 가이드를 일치시킵니다.

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Introduce a deterministic verification gate for build/test/syntax, align Claudex/Codex hook behavior and documentation around it, and add regression tests and CI to keep the harness behavior consistent across agents.

Enhancements:
- Add feature- and plan-aware L3 review detection with stronger matching signals and clearer gate messaging for WebRTC changes.
- Make Claude and Codex hooks portable by resolving project roots via git instead of hardcoded local paths.
- Document and enforce a deterministic verification protocol in agent guides and WebRTC review docs, including stricter rules against downgrading principle-level defects.
- Clarify external expert severity-handling rules to prevent unjustified downgrades of critical findings.

Build:
- Add scripts/verify-changes.sh to auto-detect changed components, run appropriate backend/frontend verification commands by risk level, and emit a reusable verification evidence block.
- Add a test harness script for verify-changes.sh and hook test sandboxes/fixtures to validate no-change, docs-only, and L1 scenarios.

CI:
- Introduce a GitHub Actions workflow to run Claude/Codex hook regression suites and verify-changes.sh smoke tests on pushes and pull requests.

Documentation:
- Add a dedicated verification protocol document and update agent/skill guides and WebRTC review protocol to require deterministic verification at Phase 03 exit and Phase 05 entry, and to record evidence before advancing or checking off TODOs.

Tests:
- Extend Claude and Codex hook test harnesses with new sandboxes and fixtures for plan/review detection, Phase 4 checklist completion, declared risk/session logs, and verification behavior edge cases.

Chores:
- Align ChatForYou external-expert guidance across Claude and Codex agents regarding severity downgrade constraints and principle-level defect handling.

</details>

### CODE REVIEW

File: .claude/hooks/capture-declared-risk.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #
 # 저장: .claude/logs/declared-risk-<session>.json  ({level, ts, excerpt})
 
-PROJECT_ROOT="/Users/sejon/project/ChatForYou_v2"
+PROJECT_ROOT="${CLAUDE_PROJECT_DIR:-$(git rev-parse --show-toplevel 2>/dev/null)}"
```
[the remark this answers is not in the document]
Comment (SeJonJ) [reply .claude/hooks/capture-declared-risk.sh:10]: 확인결과 이미 모든 경로를 컴퓨터의 개인 디렉토리가 아닌 프로젝트 디렉토리를 참고하고 있음을 확인
이슈는 잘못된 이슈로 확인되어 수정하지 않고 종료
